test/unit/math/rev/fun/log_sum_exp_test.cpp

```diff
@@ -0,0 +1,28 @@
+#include <stan/math/rev.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/math/rev/fun/util.hpp>
+#include <test/unit/math/rev/util.hpp>
+
+TEST(log_sum_exp_tests, large_values) {
+  using stan::math::var;
+
+  var a = 1e50;
+  var output = stan::math::log_sum_exp(a, a);
+  output.grad();
+  EXPECT_FLOAT_EQ(output.val(), log(2.0) + value_of(a));
+  EXPECT_FLOAT_EQ(a.adj(), 1.0);
+
+  var a1 = 1e50;
+  var a2 = 1;
```

Comment:
Once you remove the other calculate_chain thing we'll need some tests where a1 is a var and a2 is a double.